nvtabular/ops/categorify.py

```diff
@@ -201,23 +215,58 @@ def __init__(
             raise ValueError(
                 "cannot use search_sorted=True with anything else than the default freq_threshold"
             )
+        if isinstance(num_buckets, dict):
+            columns = list(num_buckets)
+            self.num_buckets = num_buckets
+        elif isinstance(num_buckets, list):
+            assert columns is not None
+            assert len(columns) == len(num_buckets)
+            self.num_buckets = {col: nb for col, nb in zip(columns, num_buckets)}
+        elif isinstance(num_buckets, int) or num_buckets is None:
+            self.num_buckets = num_buckets
+            print(self.num_buckets)
+        elif num_buckets == 0 or 0 in num_buckets:
+            raise ValueError(
+                "For hashing num_buckets should be an int > 1, otherwise set it to None."
+            )
 
     @property
     def req_stats(self):
-        return [
-            GroupbyStatistics(
-                columns=self.column_groups or self.columns,
-                concat_groups=self.encode_type == "joint",
-                cont_names=[],
-                stats=[],
-                freq_threshold=self.freq_threshold,
-                tree_width=self.tree_width,
-                out_path=self.out_path,
-                on_host=self.on_host,
-                stat_name=self.stat_name,
-                name_sep=self.name_sep,
-            )
-        ]
+        if not self.num_buckets:
+            return [
+                GroupbyStatistics(
+                    columns=self.column_groups or self.columns,
+                    concat_groups=self.encode_type == "joint",
+                    cont_names=[],
+                    stats=[],
+                    freq_threshold=self.freq_threshold,
+                    tree_width=self.tree_width,
+                    out_path=self.out_path,
+                    on_host=self.on_host,
+                    stat_name=self.stat_name,
+                    name_sep=self.name_sep,
+                )
+            ]
+        else:
+            return [
+                GroupbyStatistics(
+                    columns=self.column_groups or self.columns,
+                    concat_groups=self.encode_type == "joint",
+                    cont_names=[],
+                    stats=[],
+                    freq_threshold=self.freq_threshold,
+                    tree_width=self.tree_width,
+                    out_path=self.out_path,
+                    on_host=self.on_host,
+                    stat_name=self.stat_name,
+                    name_sep=self.name_sep,
+                ),
+                SetBuckets(
+                    columns=self.column_groups or self.columns,
+                    num_buckets=self.num_buckets,
+                    freq_limit=self.freq_threshold,
+                ),
+            ]
```

Comment:
Just a nit-pick, but it's probably best to avoid so much code duplication.

```suggestion
        stats = [
            GroupbyStatistics(
                columns=self.column_groups or self.columns,
                concat_groups=self.encode_type == "joint",
                cont_names=[],
                stats=[],
                freq_threshold=self.freq_threshold,
                tree_width=self.tree_width,
                out_path=self.out_path,
                on_host=self.on_host,
                stat_name=self.stat_name,
                name_sep=self.name_sep,
            )
        ]
        if self.num_buckets:
            stats += [
                SetBuckets(
                    columns=self.column_groups or self.columns,
                    num_buckets=self.num_buckets,
                    freq_limit=self.freq_threshold,
                )
            ]
        return stats
```

Reply:
Thanks!